electron_app/DownloadManager.js

```diff
@@ -1,29 +1,48 @@
 const Downloader = require('./Downloader');
 const path = require('path');
-const { handleFolder } = require('./helpers/handleDisk');
+const { handleFolder, fileAlreadyExists } = require('./helpers/handleDisk');
 const downloadStats = require('./helpers/downloadInfo').downloadStats;
 const handleEvents = require('./helpers/handleEvents');
 
 let filePath = '';
 let allDownloads = [];
+let fileExists = true;
 
-const downloadManager = (items, access_token, win) => {
+const downloadManager = (items, access_token, electronWin) => {
   items.forEach(item => {
-    if (item.preserveStructure) {
-      const structurePath = item.path.substring(item.path.lastIndexOf('/'), 0);
-      handleFolder(path.join(item.destination, structurePath)).then(result => {
-        allDownloads.push(processDownload(access_token, item, result, win));
-      });
-    } else {
-      allDownloads.push(processDownload(access_token, item, item.destination, win));
-    }
+    let count = 0;
+    let fileName = item.name;
+    const destination = item.preserveStructure ?
+      path.join(item.destination, item.path.substring(item.path.lastIndexOf('/'), 0)) : item.destination;
+    do {
+      fileExists = fileAlreadyExists(path.join(destination, fileName));
+      if (fileExists) {
+        count++;
+        fileName = item.name.substring(0, item.name.indexOf('.')) + '(' + (count) + ')' +
+          item.name.substring(item.name.indexOf('.'));
+      } else {
+        item.name = fileName;
+        item.destination = destination;
+        processItem(item, access_token, electronWin);
+      }
+    } while (fileExists);
   });
 };
 
+const processItem = (item, token, win) => {
+  if (item.preserveStructure) {
+    handleFolder(item.destination).then(result => {
```

Comment:
You could avoid overwriting `item.destination` above by passing in the destination directory as an argument to `processItem()`. I think you can also remove the check for `preserveStructure` here. If `preserveStructure` is false than `handleFolder()` won't need to create a directory at all, but it doesn't hurt to call it anyway.